### PR TITLE
Summarised GitHub Action workflow in one pipeline

### PR DESCRIPTION
**Pull request by Anselmoo**
GitHub Action workflows for lint, test, docs are merged in one pipeline according to issue #128

### CODE REVIEW

File: .github/workflows/ci-test.yml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+          # These packages are installed in the base environment but may be older
+          # versions. Explicitly upgrade them because they often create
+          # installation problems if out of date.
+          python -m pip install --upgrade pip setuptools numpy
+          pip install flake8
+
+      - name: Run flake8
+        run: |
+          flake8 --max-line-length=115
+  build:
```
Comment (mrakitin) [inline .github/workflows/ci-test.yml:23]: Maybe this section can be called "test"? Strictly speaking, it's building the extensions and installs the package in this job and in the `docs` job.

**Review by mrakitin — COMMENTED** [review]
I like the approach! How about reusing the separate actions and include them in this `ci-test.yml` workflow? An example https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#example-using-an-action-in-the-same-repository-as-the-workflow may help.

File: .github/workflows/ci-test.yml (hunk 1 above)
Comment (mrakitin) [reply .github/workflows/ci-test.yml:23]: Yes, I think I meant that. And the name below should reflect that.

Comment (Anselmoo) [reply .github/workflows/ci-test.yml]: I solved it, also you could keep in mind also to add the [docs-publishing](https://github.com/NSLS-II/edrixs/blob/master/.github/workflows/docs-publish.yml) in it

Comment (mrakitin) [reply .github/workflows/ci-test.yml]: Thanks!

**Comment by mrakitin** [comment]
Power-cycled to trigger CI, as it did not run.

**Comment by mrakitin** [comment]
There was a problem with the workflow after renaming the job:
image

I pushed a commit https://github.com/NSLS-II/edrixs/pull/129/commits/df1bda07d835cd013717199bba47447fdbbd7eba into your PR branch to address it.

**Review by mrakitin — APPROVED** [review]

**Pull request merged**